File: q1.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n = 5;
    int tmp = 1;
    for(int i = 0; i < n; i++){
        if(i%2 == 0){
            for(int j = 0; j < i+1; j++){
                cout<<tmp;
                if(j != i){
                    cout << "*";
                }
                tmp++;
            }
            cout <<endl;
        }else{
            int idx = tmp + i ;
            for(int j = 0; j < i+1; j++){
                cout << idx;
                if(j != i){
                    cout<<"*";
                }
                idx--;
                tmp++;
            }
            cout <<endl;
        }
    }
    return 0;
}

File: q1.java
import java.util.Scanner;

public class q1{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int tmp = 1;    
        for(int i = 0; i < n; i++){
            if(i %2 == 0){
                for(int j = 0; j < i+1; j++){
                    System.out.print(tmp);
                    if(j != i){
                        System.out.print("*");
                    }
                    tmp++;
                }
                System.out.println();
            }
            else{
                int idx = tmp + i;
                for(int j = 0; j < i+1; j++){
                    System.out.print(idx);
                    if(j != i){
                        System.out.print("*");
                    }
                    idx--;
                    tmp++;
                }
                System.out.println();
            }
        }
    }
}